SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+	GEditor->AddActor(CurrentLevel, ARPCInInterfaceTest::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart1::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart2::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, AOwnerOnlyPropertyReplication::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ADormancyAndTombstoneTest::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ASpatialTestSingleServerDynamicComponents::StaticClass(), FTransform::Identity);
+
+	// Add test helpers
+	// Unfortunately, the nature of some tests requires them to have actors placed in the level, to trigger some Unreal behavior
+	GEditor->AddActor(CurrentLevel, ADormancyTestActor::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, AReplicatedVisibilityTestActor::StaticClass(), FTransform::Identity);
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(FPaths::ProjectContentDir()
+														   + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Fast/SpatialNetworkingMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+// This map is a simple 2-server-worker map, where both workers see everything in the world
+void UGenerateTestMapsCommandlet::CreateSpatial2WorkerMap()
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the tests
+
+	// The two orchestration tests are placed at opposite points around the origin to "guarantee" they will land on different workers so
+	// that they can demonstrate they work in all situations
+	GEditor->AddActor(CurrentLevel, ACrossServerAndClientOrchestrationTest::StaticClass(), FTransform(FVector(-250, 250, 0)));
+	GEditor->AddActor(CurrentLevel, ACrossServerAndClientOrchestrationTest::StaticClass(), FTransform(FVector(250, -250, 0)));
+	// This one lives in the TestGyms, TODO
+	// GEditor->AddActor(CurrentLevel, APredictedGameplayCuesTest::StaticClass(), FTransform::Identity);
+	// This one lives in the TestGyms, TODO
+	// GEditor->AddActor(CurrentLevel, ACrossServerAbilityActivationTest::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart1::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart2::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARelevancyTest::StaticClass(), FTransform::Identity);
+
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->SetMultiWorkerSettingsClass(UTest1x2WorkerSettings::StaticClass());
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(
+		FPaths::ProjectContentDir() + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Fast_Spatial_Only/Spatial2WorkerMap.umap"));
```

Comment:
Shouldn't generated maps go in `Intermediate` directory? Won't they always pop-up in git when someone is using them with this model?

Reply:
hm, true, however, I don't think I can place them in the Intermediate directory to circumvent that. I could add a new .gitignore for these, though, I guess.

Reply:
Actually, Intermediate directory works, did that.